File: DecompiledGameScripts/cyberpunk/devices/UI/vendingMachine/imageActionButtonLogicController.swift

public class ImageActionButtonLogicController extends DeviceActionWidgetControllerBase {

  @runtimeProperty("category", "Widget Refs")
  private edit let m_tallImageWidget: inkImageRef;

  protected let m_price: Int32;

  public func Initialize(gameController: ref<DeviceInkGameControllerBase>, const widgetData: script_ref<SActionWidgetPackage>) -> Void {
    let action: ref<DispenceItemFromVendor>;
    super.Initialize(gameController, widgetData);
    action = Deref(widgetData).action as DispenceItemFromVendor;
    if IsDefined(action) {
      inkImageRef.SetTexturePart(this.m_tallImageWidget, action.GetAtlasTexture());
      this.m_price = action.GetPrice();
    };
  }

  public final func GetPrice() -> Int32 {
    return this.m_price;
  }
}

File: Definitions/Game_Definitions/classes/ImageActionButtonLogicController.lua
---@meta
---@diagnostic disable

---@class ImageActionButtonLogicController : DeviceActionWidgetControllerBase
---@field tallImageWidget inkImageWidgetReference
---@field price Int32
ImageActionButtonLogicController = {}

---@return ImageActionButtonLogicController
function ImageActionButtonLogicController.new() return end

---@param props table
---@return ImageActionButtonLogicController
function ImageActionButtonLogicController.new(props) return end

---@return Int32
function ImageActionButtonLogicController:GetPrice() return end

---@param gameController DeviceInkGameControllerBase
---@param widgetData SActionWidgetPackage
function ImageActionButtonLogicController:Initialize(gameController, widgetData) return end

